crates/loader/src/local.rs

```diff
@@ -135,6 +136,21 @@ impl LocalLoader {
         })
     }
 
+    fn env_checker((key, val): (String, Variable)) -> anyhow::Result<(String, Variable)> {
+        if val.default.is_none() {
+            if std::env::var(env_key(None, key.as_ref())).is_err() {
```

Comment:
You can't assume that variables will come from the host environment.  They could come from Vault or Azure KeyVault - for example passwords or tokens would typically be stashed in vaults and it would be rude to demand users create dummy EVs for them.  Variable sources are defined in the runtime config file - you can't validate them until you have the runtime config providers lined up.